### PR TITLE
add high level API to set priority of system handlers

### PR DESCRIPTION
**Pull request by japaric**
needed for cortex-m-rtfm v0.4.x (it makes my life easier / it makes it easier to
support ARMv6-M)

r? @rust-embedded/cortex-m (anyone)

### CODE REVIEW

File: src/peripheral/scb.rs [hunk 1]
```diff
@@ -669,3 +669,116 @@ impl SCB {
...
+    UsageFault,
+
+    /// Secure fault interrupt (only on ARMv8-M)
+    #[cfg(any(armv8m, target_arch = "x86_64"))]
+    SecureFault,
+
+    /// SV call interrupt
+    SVCall,
+
+    // #[cfg(not(armv6m))]
```
Comment (therealprof) [inline src/peripheral/scb.rs]: I'm not so hot about having this commented out bit in here. Can we figure out whether it has configurable priority and either take it in or leave it out?

Comment (adamgreig) [reply src/peripheral/scb.rs]: It's got `PRI_12` in `SHPR3` too:
image

Comment (japaric) [reply src/peripheral/scb.rs]: @adamgreig Thanks. PRI_12 was omitted in the Cortex-M4 documentation I read. (Not the first time ARM omits information about debugging / profiling capabilities in some revisions of their docs)

**Comment by japaric** [comment]
Addressed review comments

**Comment by therealprof** [comment]
bors try

**Pull request merged**